src/domain/covenant-identity.ts

```diff
@@ -0,0 +1,429 @@
+import {
+  Identity,
+  crypto,
+  address,
+  Psbt,
+  restorerFromEsplora,
+  IdentityType,
+  restorerFromState,
+  Transaction,
+  bip341,
+  toXpub,
+  witnessStackToScriptWitness,
+  checkIdentityType,
+  checkMnemonic,
+  networks,
+  fromXpub,
+} from 'ldk';
+import type {
+  AddressInterface,
+  IdentityInterface,
+  IdentityOpts,
+  Restorer,
+  EsploraRestorerOpts,
+  NetworkString,
+  StateRestorerOpts,
+  Mnemonic,
+} from 'ldk';
+import type { BlindingDataLike } from 'liquidjs-lib/src/psbt';
+import { evaluate } from '../descriptors';
+import type { Result } from '../descriptors/ast';
+import type { Context } from '../descriptors/preprocessing';
+import { SLIP77Factory } from 'slip77';
+import * as ecc from 'tiny-secp256k1';
+import type { BIP32Interface } from 'bip32';
+import BIP32Factory from 'bip32';
+import { mnemonicToSeedSync } from 'bip39';
+import type { ScriptInputsNeeds } from './script-analyser';
+import { analyzeTapscriptTree } from './script-analyser';
+
+// slip13: https://github.com/satoshilabs/slips/blob/master/slip-0013.md#hd-structure
+function namespaceToDerivationPath(namespace: string): string {
+  const hash = crypto.sha256(Buffer.from(namespace));
+  const hash128 = hash.slice(0, 16);
+  const A = hash128.readUInt32LE(0) || 0x80000000;
+  const B = hash128.readUint32LE(4) || 0x80000000;
+  const C = hash128.readUint32LE(8) || 0x80000000;
+  const D = hash128.readUint32LE(12) || 0x80000000;
+  return `m/${A}/${B}/${C}/${D}`;
+}
+
+function makeBaseNodeFromNamespace(m: BIP32Interface, namespace: string): BIP32Interface {
+  const path = namespaceToDerivationPath(namespace);
+  return m.derivePath(path);
+}
+
+export interface TaprootAddressInterface extends AddressInterface {
+  result: Result;
+  tapscriptNeeds: Record<string, ScriptInputsNeeds>; // scripthex -> needs
+}
+
+export interface CovenantDescriptors {
+  namespace: string;
+  template?: string;
+  changeTemplate?: string;
+  isSpendableViaUI?: boolean;
```

Comment:
```suggestion
  isSpendable?: boolean;
```

Reply:
isSpendable may mislead future devs IMO, what do u think of `isSpendableByMarina` ?

Reply:
Yep better